### PR TITLE
fix: bug #2614 class-name is now matched before keyword

### LINKED ISSUE #2614: [PHP] When code begins with class, class-name after it is not matched.
**Information**
- Language: PHP
- Plugins: none

**Description**
When code begins with `class | interface | trait` the class-name after it is not matched.
When an whitespace is added before `class` the class-name is matched correctly.

I know that correct php code starts with `<?php` and then first class-name is matched correctly.
But opening tag is often dropped in code snippets fot better readability (blogs, docs, ..) and then this is an issue.

**Code snippet**

<details>
<summary>The code being highlighted incorrectly. Class name `First` is not matched.</summary>

```
class First {}
class Second {}
```

</details>

### PR DESCRIPTION
**Pull request by TomPavelec**
Closes #2614

### CODE REVIEW

File: components/prism-php.js [hunk 1]
```diff
@@ -33,6 +33,14 @@
 				'punctuation': /\\/
 			}
 		},
+		'class-name-definition': [
+			{
+				pattern: /(\b(?:class|interface|trait)\s+)\b[a-z_]\w*(?!\\)\b/i,
+				greedy: true,
```
Comment (RunDevelopment) [inline components/prism-php.js:39]: I don't think greedy is necessary, is it?

File: components/prism-php.js [hunk 2]
```diff
@@ -33,6 +33,14 @@
 				'punctuation': /\\/
 			}
 		},
+		'class-name-definition': [
```
Comment (RunDevelopment) [inline components/prism-php.js:36]: The array isn't necessary.

File: components/prism-php.js [hunk 3]
```diff
@@ -33,6 +33,14 @@
...
+			{
+				pattern: /(\b(?:class|interface|trait)\s+)\b[a-z_]\w*(?!\\)\b/i,
```
Comment (RunDevelopment) [inline components/prism-php.js:38]: With `class|interface|trait` covered, you can remove them from the first `class-name` pattern. 

(GH doesn't allow me to make a comment there, so I'm making it here.)

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for the PR @TomPavelec!

A few nits. Also, could you please add a test for the code snippet in #2614 to show that the bug has been fixed?

File: components/prism-php.js (hunk 2 above)
Comment (TomPavelec) [reply components/prism-php.js]: Fixed

File: components/prism-php.js (hunk 3 above)
Comment (TomPavelec) [reply components/prism-php.js]: Removed

File: components/prism-php.js (hunk 1 above)
Comment (TomPavelec) [reply components/prism-php.js]: Yes, I think that greedy can be dropped.

**Comment by TomPavelec** [comment]
Thank you for quick response @RunDevelopment.

**Review by RunDevelopment — APPROVED** [review]

**Comment by RunDevelopment** [comment]
Thank you for contributing @TomPavelec!

**Pull request merged**